src/background/requestHandler.js

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {Component} from "./component.js";
+import {Logger} from "./logger.js";
+
+const log = Logger.logger("RequestHandler");
+
+let self;
+
+export class RequestHandler extends Component {
+  constructor(receiver) {
+    super(receiver);
+
+    self = this;
+    
+  }
+
+  init() {
```

Comment:
Ah so i just slapped typescript typechecking on my branch. 
This needs to be async init() otherwise its not a good overwrite.